### DISCUSSION TITLE
How do I unset the expecting reply bit?

### DISCUSSION THREAD

**Discussion by djlad** [JoelBender/bacpypes#471] [Q&A]
Hi, I'm trying to unset the expecting reply bit in a ReadPropertyMultipleRequest to troubleshoot a device.
```
        request = ReadPropertyMultipleRequest(listOfReadAccessSpecs=readAccessSpecList)
        request.pduExpectingReply = 0
        request.pduDestination = Address(address)
        iocb = IOCB(request)
        deferred(self.request_io, iocb)
```
Is there something I'm missing? The expecting reply bit is still set.

I'm trying this because I'm occasionally getting abort reject reason 65 (noResponse) when I read from a "PurpleSwift bacnet_p2 gateway". I noticed that PurpleSwift's bacnet explorer reliably reads every time and doesn't have the expecting reply bit set.

Also, does reason 65 imply a timeout? Is it possible to increase the time before it times out?

**Comment by JoelBender** [comment]
Second question is easier, the 65 is an application layer timeout, so the APDU timeout and retry count has been exceeded.  You can change those parameters by providing **apduTimeout** and **numberOfApduRetries** values in the **LocalDeviceObject** object that is created, the defaults of 3000 milliseconds and 3 are in the bacpypes.local.device module.

The first question is harder, I can confirm that turning off `request.pduExpectingReply` in your application doesn't propagate down so when the PDU hits the UDP communications virtual link layer it has been turned back on again.  Humph.

**Reply by JoelBender** [reply]
Issue #472 created.  A Wireshark pcap file with the working requests from the BACnet explorer and your not-working requests might reveal other problems.

**Reply by JoelBender** [reply]
When I was confirming there was a problem it turns out I was looking at the wrong bit.  The gist [here](https://gist.github.com/JoelBender/40349dfc93a35944e131ce8494f902c6) shows the stack is working correctly.
